### PR TITLE
docs: sync hooks and waves

### PR DESCRIPTION
**Pull request by kostis-codefresh**
Signed-off-by: Kostis Kapelonis <kostis@codefresh.io>

Ongoing work for https://github.com/argoproj/argo-cd/issues/5635

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/syncing/waves.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      restartPolicy: Never
+  backoffLimit: 1
+```
+
+For more information at possible scenarios look at [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/job/) and [Argo Workflows](https://argoproj.github.io/workflows/).
+
+
+
+
+
```
Comment (terrytangyuan) [inline docs/syncing/waves.md:248]: Remove redundant spaces

**Review by terrytangyuan — APPROVED** [review]
Looks great!

**Comment by jannfis** [comment]
Another great chapter, @kostis-codefresh - thank you so much!

I was wondering, now as this part will cover sync hooks, if the `syncing/hooks` chapter (which is empty right now) could be removed? I think that you have put hooks specifics into sync waves/phases chapter is the right thing to do.

**Review by iam-veeramalla — APPROVED** [review]
Great documentation of Sync waves and hooks with examples. Thanks @kostis-codefresh !!

File: docs/syncing/waves.md (hunk 1 above)
Comment (kostis-codefresh) [reply docs/syncing/waves.md]: Removed

**Comment by kostis-codefresh** [comment]
@jannfis You are right. I removed the hooks chapter as you suggested.

@terrytangyuan I removed the spaces

This is ready to be merged now, if you don't have any other comments.

**Review by jannfis — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**